src/Arcus.Templates.Tests.Integration/Fixture/WebApiProjectOptions.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.IO;
+using Arcus.Security.Secrets.Core.Interfaces;
+using GuardNet;
+
+namespace Arcus.Templates.Tests.Integration.Fixture
+{
+    /// <summary>
+    /// Represents all the available project options on the web API template project.
+    /// </summary>
+    public class WebApiProjectOptions : ProjectOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebApiProjectOptions"/> class.
+        /// </summary>
+        public WebApiProjectOptions()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebApiProjectOptions"/> class.
+        /// </summary>
+        private WebApiProjectOptions(ProjectOptions options) : base(options)
+        {
+        }
+
+        /// <summary>
+        /// Creates a set of empty project options for the web API project; resulting in a default web API project when a project is created from these options.
+        /// </summary>
+        public static readonly WebApiProjectOptions Empty = new WebApiProjectOptions();
+
+        /// <summary>
+        /// Adds a shared access key authentication to the web API project.
+        /// </summary>
+        /// <param name="headerName">The name of the request header which value must match the stored secret.</param>
+        /// <param name="secretName">The name of the secret that's being retrieved using the <see cref="ISecretProvider.Get"/> call.</param>
+        /// <param name="secretValue">The value of the secret that should be retrieved using the <see cref="ISecretProvider.Get"/> call.</param>
+        public WebApiProjectOptions WithSharedAccessAuthentication(string headerName, string secretName, string secretValue)
+        {
+            Guard.NotNullOrWhitespace(headerName, nameof(headerName), "Cannot add shared access key authentication project option without a HTTP request header name containing the secret name");
+            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Cannot add shared access key authentication project option without a secret name");
+            Guard.NotNullOrWhitespace(secretValue, nameof(secretValue), "Cannot add shared access key authentication project option without a secret value");
+
+            ProjectOptions optionsWithSharedAccessAuthentication = AddOption(
+                "--Authentication SharedAccessKey",
+                projectDirectory => ConfigureSharedAccessAuthentication(projectDirectory, headerName, secretName, secretValue));
+
+            return new WebApiProjectOptions(optionsWithSharedAccessAuthentication);
+        }
+
+        private void ConfigureSharedAccessAuthentication(DirectoryInfo projectDirectory, string requestHeader, string secretName, string secretValue)
```

Comment:
This looks hacky but probably have to use it? 😁

Reply:
Yeah, and I guess that for some other features; we also would have to alter the created project in a certain way so it is usable.

This code to add a stub secret provider can be reused though if we want to test the certificate authentication.